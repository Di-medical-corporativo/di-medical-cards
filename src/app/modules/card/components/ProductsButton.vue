<template>
  <button
    class="card-contact__products__button"
    :style="buttonBackground"
    @click="goToProducts()"
  >
    <i class="bi bi-cart-fill icon"></i>
    Produtos
  </button>
</template>

<script>
import { computed } from 'vue'
import { useRouter } from 'vue-router'

export default {
  props: {
    sucursal: {
      type: String,
      required: true
    }
  },

  setup (props) {
    const router = useRouter()
    return {
      buttonBackground: computed(() => {
        const backgroundSucursal =
          props.sucursal === 'sur' ? '#218d9b' : '#c28400'
        return { background: backgroundSucursal }
      }),
      goToProducts: () => router.push({ name: 'products-card' })
    }
  }
}
</script>
