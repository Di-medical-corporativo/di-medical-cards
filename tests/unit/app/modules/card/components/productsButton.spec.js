import { shallowMount } from '@vue/test-utils'
import ProductsButton from '../../../../../../src/app/modules/card/components/ProductsButton.vue'

jest.mock('vue-router', () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn()
  }))
}))

describe('Products button to catalogue', () => {
  test('should match snapshot', () => {
    const wrapper = shallowMount(ProductsButton, {
      propsData: {
        sucursal: 'sur'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should change color background depending on sucursal, sur case', () => {
    const wrapper = shallowMount(ProductsButton, {
      propsData: {
        sucursal: 'sur'
      }
    })
    expect(wrapper.vm.buttonBackground.background).toBe('#218d9b')
  })

  test('should change color background depending on sucursal, corporativo case', () => {
    const wrapper = shallowMount(ProductsButton, {
      propsData: {
        sucursal: 'corporativo'
      }
    })
    expect(wrapper.vm.buttonBackground.background).toBe('#c28400')
  })

  test('should have cart icon', () => {
    const wrapper = shallowMount(ProductsButton, {
      propsData: {
        sucursal: 'corporativo'
      }
    })
    const icon = wrapper.get('i')
    const iconName = 'bi bi-cart-fill icon'
    expect(icon.attributes('class')).toBe(iconName)
  })

  test('should call goToProducts when the button is clicked', async () => {
    const wrapper = shallowMount(ProductsButton, {
      propsData: {
        sucursal: 'corporativo'
      }
    })
    const spy = (wrapper.vm.goToProducts = jest.fn())
    await wrapper.get('button').trigger('click')
    expect(spy).toHaveBeenCalled()
  })
})
