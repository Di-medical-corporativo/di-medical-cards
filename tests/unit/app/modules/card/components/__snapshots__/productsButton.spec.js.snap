// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products button to catalogue should match snapshot 1`] = `<button class="card-contact__products__button" style="background: rgb(33, 141, 155);"><i class="bi bi-cart-fill icon"></i> Produtos </button>`;
